projects/wave-core-new/src/lib/time/time-slider/time-slider.component.ts

```diff
@@ -0,0 +1,282 @@
+                // doing it the momentjs locale aware way
+                clone.weekday(0);
+                clone.add(1, 'week');
+            } else {
+                clone.weekday(0);
+            }
+
+            clone.hours(0);
+            clone.minutes(0);
+            clone.seconds(0);
+            clone.milliseconds(0);
+        } else if (scale === 'day') {
+            //noinspection FallthroughInSwitchStatementJS
+            switch (step) {
+                case 5:
+                case 2:
+                    clone.hours(Math.round(clone.hours() / 24) * 24);
+                    break;
+                default:
+                    clone.hours(Math.round(clone.hours() / 12) * 12);
+                    break;
+            }
+            clone.minutes(0);
+            clone.seconds(0);
+            clone.milliseconds(0);
+        } else if (scale === 'weekday') {
+            //noinspection FallthroughInSwitchStatementJS
+            switch (step) {
+                case 5:
+                case 2:
+                    clone.hours(Math.round(clone.hours() / 12) * 12);
+                    break;
+                default:
+                    clone.hours(Math.round(clone.hours() / 6) * 6);
+                    break;
+            }
+            clone.minutes(0);
+            clone.seconds(0);
+            clone.milliseconds(0);
+        } else if (scale === 'hour') {
+            switch (step) {
+                case 4:
+                    clone.minutes(Math.round(clone.minutes() / 60) * 60);
+                    break;
+                default:
+                    clone.minutes(Math.round(clone.minutes() / 30) * 30);
+                    break;
+            }
+            clone.seconds(0);
+            clone.milliseconds(0);
+        }
+        return clone;
+    }
+
+    //every group represents one layer
+    getTimelineGroups(): void {
+        this.groups = new DataSet([]);
+        for (const layer of this.layerList) {
+            this.groups.add({
+                id: layer.id,
+                content: layer.name,
+            });
+        }
+    }
+
+    //every group (layer) can hav multiple data points
+    //ToDo: implement the availability of the layers
+    getTimelineData(): void {
+        this.data = new DataSet();
+        for (const layer of this.layerList) {
+            this.data.add({
+                id: layer.id,
+                group: layer.id,
+                content: '',
+                //     type: 'point' or 'background',
+                start: '2012-01',
+                end: '2025-01',
+            });
+        }
+    }
+
+    //options for the timeline
+    getOptions(): void {
+        this.options = {
+            moment: (date: Date) => moment(date).utc(), //use utc
+            snap: (date: Date, scale: string, step: number): Moment => this.snapFunction(date, scale, step),
+            start: '2012-01',
+            end: '2020-01',
+            orientation: 'top',
+            height: 150,
```

Comment:
Can't we make this an input for this component? Otherwise, we can only use it for 150px all the time.

Reply:
I made it an input. I guess thats the better solution: d972aed5751a3da301a105572fae7c632848860e
and 5c766b4e3ac8b59f4d8710bae78a38a27b416590